File: stepik-java-api/src/main/java/org/stepik/api/queries/steps/StepikStepsGetQuery.java
package org.stepik.api.queries.steps;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.steps.Steps;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikStepsGetQuery extends StepikAbstractGetQuery<StepikStepsGetQuery, Steps> {
    public StepikStepsGetQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Steps.class);
    }

    @NotNull
    public StepikStepsGetQuery page(int value) {
        addParam("page", value);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.STEPS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/steps/StepikStepsGetQuery.kt
package org.stepik.api.queries.steps

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.steps.Steps
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikStepsGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikStepsGetQuery, Steps>(stepikAction, Steps::class.java) {
    
    fun page(value: Int): StepikStepsGetQuery {
        addParam("page", value)
        return this
    }
    
    fun lesson(value: Long): StepikStepsGetQuery {
        addParam("lesson", value)
        return this
    }
    
    override val url = "${this.stepikAction.stepikApiClient.host}/api/steps"
    
}
